tests/rollback/setup.yml

```diff
@@ -0,0 +1,22 @@
+---
+- name: setup
+  hosts: ofn_servers
+  remote_user: "{{ user }}"
+
+  tasks:
+    - name: setup | rollback | add testable entry to database
+      become: yes
+      become_user: postgres
+      command: |
+        psql {{ db }} -c "
+        UPDATE spree_preferences
+        SET value='rollback_test'
+        WHERE key='spree/app_configuration/privacy_policy_url';"
```

Comment:
This row would initially not exist, right?

Reply:
It's a standard spree preference, so the row will exist at this point, but we're setting a non-default value of `rollback_test`.

Reply:
P.S. the build is green, so you can see the Ansible output already in the Semaphore build for this PR. :point_down:

Reply:
Is this created somewhere in the setup file aside from `script/setup`? I asked because `script/setup` (running Rake task `ofn:sample_data`) does not create a `Spree::Preference` record for this.

Is this why the check passes?

```
failed_when: rollback_test.stdout == '1'
```

Sorry if I'm missing something supposedly obvious here.

Reply:
@Matt-Yorkley I have a pending small concern here. I believe the `Spree::Preference` here does not actually exist yet and the `failed_when` below is wrong (`stdout` output should be 1 if there is a match).

But already happy for this PR to be merged if that is not the case.

Reply:
@Matt-Yorkley I have a pending small concern here. I believe the `Spree::Preference` here does not actually exist yet and the `failed_when` below is wrong (`stdout` output should be 1 if there is a match).

But already happy for this PR to be merged if that is not the case.

Reply:
The `UPDATE` query passes in the build:

![exists](https://user-images.githubusercontent.com/9029026/58429926-540ffc80-809f-11e9-8a88-a4388de1194d.png)

It would fail if that record didn't exist, right?

Reply:
This setup step takes place after a full provision and deploy. Those Spree preferences are stored in the database by that point.

Reply:
@Matt-Yorkley A SQL `UPDATE` query does not fail if there is no record to update, and not all Spree preferences are inserted in the database when `script/setup` is run. The Spree preference for privacy policy URL is one the many that are not created - it is just un-set.

Reply:
Ah you're right! I must have looked in my dev database instead of my ofn-install Vagrant database when choosing a key. I've changed it to an `INSERT` of a unique record.

Reply:
We're still doing `failed_when: rollback_test.stdout == '1'` in the test though, as the record should _not_ exist after the rollback playbook has run.

Reply:
That sounds right, @Matt-Yorkley! :slightly_smiling_face: